src/expr/src/scalar/func/impls/string.rs

```diff
@@ -205,6 +205,20 @@ sqlfunc!(
     }
 );
 
+sqlfunc!(
+    #[sqlname = "try_parse_monotonic_iso8601_timestamp"]
+    // TODO: Pretty sure this preserves uniqueness, but not 100%.
+    #[is_monotone = true]
```

Comment:
Sadly this is not quite right, because of the nulls - we can't assume that `parse(A) < parse(B)` given `A < B`, because `parse(A)` might be null.

This is not really a problem for the approach... we just need to assume that we might return null for any nontrivial range of strings. But it does mean that this needs special-case support in `interpret.rs`, instead of getting to use this annotation. (Like `SpecialBinary`, but unary.)

Annoying, but hopefully not more so than the annoyances you have already surmounted! Feel free to ping me if anything I'm saying doesn't make sense!

Reply:
still working on this

Reply:
done!